angular-workspace/projects/ni/nimble-angular/label-provider/rich-text/tests/nimble-label-provider-rich-text.directive.spec.ts

```diff
@@ -0,0 +1,283 @@
+import { Component, ElementRef, ViewChild } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { LabelProviderRichText, NimbleLabelProviderRichTextDirective } from '../nimble-label-provider-rich-text.directive';
+import { NimbleLabelProviderRichTextModule } from '../nimble-label-provider-rich-text.module';
+
+describe('Nimble Label Provider Rich text', () => {
+    const label1 = 'String 1';
+    const label2 = 'String 2';
+    const label3 = 'String 3';
+    const label4 = 'String 4';
```

Comment:
await: Always have meaningful names to represent the different labels we are going to test here. For ex: `toggleBoldLabel`, `toggleItalicsLabel`, etc. 

As a general thumb rule, take time to step back and refactor the file where you're pasting the code to avoid any copy paste bias before submission of any PR. 

Make sure that you are considering to use `getSpecTypeByNamedList` throughout this file:

1. Consider putting these individual constants into an array of constant objects (`richTextLabels`) with necessary metadata like `name`, `value` etc. and type it as `RichTextLabel[]`.
2. After creating an array, consider using `getSpecTypeByNamedList` to iterate through `RichTextLabel[]` and insert label name in the `it` test case and use `value` inside test cases for your assertion. I think many of the describe blocks below can eliminate many lines of code with this approach.

Reply:
Looked into the `angular-workspace` for test utilities like `getSpecTypeByNamedList` and it seems like that is available only under `nimble-components`. So just added separate assertions for all the checks.

Would like to get the thoughts of nimble folks

Reply:
Sure. Pls make sure that you are changing the `label1`, `label2` to meaningful names then.

Reply:
The names `label1`, `label2` etc. are the different labels used to assert the change, and it not specific to `toggleBold` or `toggleItalics` functionalities. So making it as `toggleBoldLabel` or some functional specific name will be meaning different in a case where unless we assign `label1` to the `toggleItalics` property and reassign the same property with another label in this case I named it as `label2` . 

Like to get your thoughts on this.

Reply:
This is little nit picky to just improve the readability. `toggleBold` was an example. I hope this example can suit in this way. Do you have any other suggestions - 

```suggestion
    const boldLabel = 'Bold';
    const italicsLabel= 'Italics';
    const numberedListLabel= 'Numbered List';
    const bulletedListLabel = 'Bulleted List';
```

The corresponding test cases can have `newBoldLabel`, `newItalicsLabel`, etc. to assert?

Reply:
Updated with specific labels and asserted accordingly